### ISSUE TITLE
Performance issues with projects view when you have lots of sub projects

### ISSUE THREAD

**Issue by fellhorn** [allegroai/clearml-web#32]
We see some performance issues with the projects list in the Web UI. When there are lots of sub projects the server response from `projects.get_all_ex grows` to > 1 MB for us and takes a couple of seconds. Mainly because of the `"sub_projects"` field of each top level project in the response.

We were wondering whether the sub projects list could be deactivated completely (the server does not seem to support that yet as it always sends the `sub_projects` when one requests the statistics).

Alternatively it would be much more efficient for us to only list the next hierarchy instead of all sub projects as our project tree is deeply nested.

**Comment by shyallegro** [comment]
Hi @fellhorn, I was wandering what this huge hierarchy models? and whether tagging experiments can replace this practice, till we optimize the projects view for your special use case.

**Comment by fellhorn** [comment]
Thanks for coming back to us. I agree that our use case is probably rather special:
We also upload certain assets like checkpoints and logs to an object storage (e.g. S3, GCS, ...). We have a certain fixed path/folder structure there, which we also wanted to mirror in ClearML. That makes it easier to find a run in the object storage in ClearML and vice versa as the URLs can be easily mapped.

This is of course something we are rethinking as the we see major performance issues with it.
On the other side: it would be completely fine for us to only load one level of the tree at a time. The quick navigation and sub project counter does not give us any value
![image](https://user-images.githubusercontent.com/26092524/186129106-3ce34856-f548-4161-9a67-6c74e1ba8ad2.png)

Maybe that makes it easier to optimize as you could only load the direct childs of the current level?

**Comment by fellhorn** [comment]
Thanks for directly coming back to us. I attached the [apiserver.log](https://github.com/allegroai/clearml-web/files/9423872/apiserver.log).

The slowest API calls seem to be to `projects.get_all_ex`, e.g.
```bash
projects.get_all_ex in 4776ms
projects.get_all_ex in 5175ms
```
This alternative mapping also went through our head, we just wanted to avoid always having to go to ClearML first to find the other data in the bucket. Right now the mapping works in both ways and both the bucket and ClearML can be explored. Though if there is no other solution this would be of course an alternative.

Of course I also understand that the full list might be useful to some people. I can just imagine that this could be near impossible to implement efficiently if one has a large project tree. You basically loose all performance gains from pagination as you always query the entire tree.

**Comment by NathanClearML** [comment]
Hi @fellhorn , we couldn't replicate it on our end. Can you check if the issue persists? We also improved subprojects list behavior. 

If long response time issue persists - what scope of projects are we talking about? How many projects with many subprojects show in the recent projects view?

**Comment by fellhorn** [comment]
Hi @NathanClearML, thanks for coming back to that topic!

I just updated to the newest version of ClearML and we already saw a slight improvement over the last months. Currently the requests that previously took ~5 sec as posted [here](https://github.com/allegroai/clearml-web/issues/32#issuecomment-1227091055), now take about 2-3 sec. Definitively a huge improvement in usability. Thank you.

On your other question: The initial projects view before clicking "Load more", downloads the information on 11 projects with a total of  ~4k sub projects and ~7k experiments for us. The uncompressed response of `projects.get_all_ex`  is 450 kB.
